Add teams to group datasource

eochoa260 opened the pull request.

eochoa260 commented on opslevel/datasource_opslevel_group.go:
```diff
@@ -52,6 +52,22 @@ func datasourceGroup() *schema.Resource {
+			"teams": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
```
Now that I've implemented the resource, it seems like we would want this to be a list similar to what's in the resource schema instead of a map.  Can't remember why we decided to do maps for parent and members though.  I feel like I would want to just reference data.opslevel_group.foo.teams and it would be the same type as what the resource is expecting, e.g. list of string aliases.

rocktavious replied:
Ya that would be a good improvement.

rocktavious commented:
LGTM - Up to you @eochoa260 if you want to take it further and build it as a list.  This gets the job done but better UX is always something to strive towards.

eochoa260 commented:
@rocktavious yeah, just went ahead and updated it so that it would match what I implemented on the resource side.  A bit cleaner now and just tested that it works as expected.

rocktavious reviewed (approved):
LGTM
